internal/services/appservice/static_web_app_link_backend.go

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package appservice
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonids"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/apimanagement/2022-08-01/apimanagementservice"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/containerapps/2023-05-01/containerapps"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/web/2023-01-01/staticsites"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type StaticWebAppLinkBackendResource struct{}
+
+var _ sdk.Resource = StaticWebAppLinkBackendResource{}
+
+type StaticWebAppLinkBackendModel struct {
+	StaticWebAppID    string `tfschema:"static_web_app_id"`
+	BackendResourceId string `tfschema:"backend_resource_id"`
+}
+
+type linkedBackend struct {
+	id   string
+	name string
+}
+
+func (r StaticWebAppLinkBackendResource) ResourceType() string {
+	return "azurerm_static_web_app_link_backend"
+}
+
+func (r StaticWebAppLinkBackendResource) ModelObject() interface{} {
+	return &StaticWebAppLinkBackendModel{}
+}
+
+func (r StaticWebAppLinkBackendResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return staticsites.ValidateLinkedBackendID
+}
+
+func (r StaticWebAppLinkBackendResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"static_web_app_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: staticsites.ValidateStaticSiteID,
+		},
+
+		"backend_resource_id": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.Any(
+				apimanagementservice.ValidateServiceID,
+				commonids.ValidateAppServiceID,
+				containerapps.ValidateContainerAppID,
+			),
+		},
+	}
+}
+
+func (r StaticWebAppLinkBackendResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"id": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
```

Comment:
id is handled implicitly by the SDK package and cannot be specified here
```suggestion
	return map[string]*pluginsdk.Schema{}
```